internal/container/container.go

```diff
@@ -0,0 +1,249 @@
+package container
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"os"
+	"strconv"
+	"time"
+
+	"github.com/docker/docker/api/types"
+	"github.com/docker/docker/api/types/container"
+	"github.com/docker/docker/api/types/filters"
+	"github.com/docker/docker/client"
+	"github.com/docker/go-connections/nat"
+
+	"github.com/TBD54566975/ftl/internal/log"
+)
+
+func newClient() (*client.Client, error) {
+	cli, err := client.NewClientWithOpts(client.FromEnv, client.WithAPIVersionNegotiation())
+	if err != nil {
+		return nil, fmt.Errorf("failed to create Docker client: %w", err)
+	}
+
+	return cli, nil
+}
+
+func DoesExist(ctx context.Context, name string) (bool, error) {
+	cli, err := newClient()
+	if err != nil {
+		return false, err
+	}
+
+	containers, err := cli.ContainerList(ctx, container.ListOptions{
+		All:     true,
+		Filters: filters.NewArgs(filters.Arg("name", name)),
+	})
+	if err != nil {
+		return false, fmt.Errorf("failed to list containers: %w", err)
+	}
+
+	return len(containers) > 0, nil
+}
+
+// Run starts a new detached container with the given image, name, port map, and (optional) volume mount.
+func Run(ctx context.Context, image, name string, hostPort, containerPort int, volume string) error {
```

Comment:
Use `optional.Option[string]` for optional types - we're trying to use these everywhere we want to indicate optionality rather than using zero values/pointers.